integrations/GA4/browser.js

```diff
@@ -0,0 +1,211 @@
+              // To Do Future Scope :: implement using recursion to handle multi level prop mapping
+              if (result.length > 1) {
+                const levelOne = result[0];
+                const levelTwo = result[1];
+                item[levelTwo] = props[key];
+                if (!destinationProperties[levelOne]) {
+                  destinationProperties[levelOne] = [];
+                  destinationProperties[levelOne].push(item);
+                }
+              } else {
+                destinationProperties[result] = props[key];
+              }
+            });
+          } else {
+            destinationProperties[param.dest] = props[key];
+          }
+        }
+      });
+    });
+    return destinationProperties;
+  }
+
+  getDestinationItemProperties(products, item) {
+    const items = [];
+    let obj = {};
+    products.forEach((p) => {
+      obj = {
+        ...this.getDestinationEventProperties(p, itemParameter),
+        ...(item && item[0]),
+      };
+      items.push(obj);
+    });
+    return items;
+  }
+
+  track(rudderElement) {
+    let { event } = rudderElement.message;
+    const { properties } = rudderElement.message;
+    const { products } = properties;
+    let destinationProperties = {};
+    if (!event || this.isReservedName(event)) {
+      throw Error("Cannot call un-named/reserved named track event");
+    }
+
+    const obj = this.getDestinationEvent(event);
+    if (obj) {
+      if (products && Array.isArray(products)) {
+        event = obj.dest;
+        // eslint-disable-next-line no-const-assign
+        destinationProperties = this.getDestinationEventProperties(
+          properties,
+          eventParameter
+        );
+        destinationProperties.items = this.getDestinationItemProperties(
+          products,
+          destinationProperties.items
+        );
+      } else {
+        event = obj.dest;
+        if (!obj.hasItem) {
+          // eslint-disable-next-line no-const-assign
+          destinationProperties = this.getDestinationEventProperties(
+            properties,
+            eventParameter
+          );
+        } else {
+          // create items
+          destinationProperties.items = this.getDestinationItemProperties([
+            properties,
+          ]);
+        }
+      }
+    } else {
+      destinationProperties = properties;
+    }
+
+    this.sendGAEvent(event, destinationProperties);
+  }
+
+  identify(rudderElement) {
+    if (this.sendUserId && rudderElement.message.userId) {
+      const userId = this.analytics.userId || this.analytics.anonymousId;
+      window.gtag("config", this.measurementId, {
+        user_id: userId,
+      });
+    }
+    window.gtag("set", "user_properties", this.analytics.userTraits);
+    logger.debug("in GoogleAnalyticsManager identify");
+  }
+
+  page(rudderElement) {
+    window.gtag(
+      "event",
+      "page_view",
+      (rudderElement.message.context && rudderElement.message.context.page) ||
```

Comment:
https://support.google.com/analytics/answer/9234069
let's try to map some of our params to the params mentioned in this link,
For ex: page_view ==> 

- page_location
- page_referrer
- page_title
(So that we don't blow up number of unique params supported by GA..there is a limit there I think of unique params)
For extra params let's take a boolean input from config like send extra_params for page view and then send them. Also use rudderElement.message.properties instead of rudderElement.message.context.page

Reply:
Also check if GA4 allows sending campaign info present under rudderElement.message.context.campaign

Reply:
Added function to support GA4 params and added config to decide if need to send extra params